llama_stack/apis/eval/eval.py

```diff
@@ -85,37 +112,42 @@ async def evaluate_rows(
         candidate: EvalCandidate,
     ) -> EvaluateResponse:
         """Evaluate a list of rows on a candidate.
-        
+
         :param dataset_rows: The rows to evaluate.
         :param scoring_fn_ids: The scoring function ids to use for the evaluation.
         :param candidate: The candidate to evaluate on.
         :return: EvaluateResponse object containing generations and scores
         """
 
-    @webmethod(route="/eval/benchmarks/{benchmark_id}/jobs/{job_id}", method="GET")
-    async def job_status(self, benchmark_id: str, job_id: str) -> Optional[JobStatus]:
-        """Get the status of a job.
+    @webmethod(route="/eval/jobs", method="GET")
+    async def list_eval_jobs(self) -> List[EvalJob]:
+        """List all evaluation jobs.
 
-        :param benchmark_id: The ID of the benchmark to run the evaluation on.
-        :param job_id: The ID of the job to get the status of.
-        :return: The status of the evaluationjob.
+        :return: A list of evaluation jobs.
         """
         ...
 
-    @webmethod(route="/eval/benchmarks/{benchmark_id}/jobs/{job_id}", method="DELETE")
-    async def job_cancel(self, benchmark_id: str, job_id: str) -> None:
-        """Cancel a job.
+    @webmethod(route="/eval/jobs/{job_id}", method="GET")
+    async def get_eval_job(self, job_id: str) -> Optional[EvalJob]:
+        """Get a job by id.
 
-        :param benchmark_id: The ID of the benchmark to run the evaluation on.
-        :param job_id: The ID of the job to cancel.
+        :param job_id: The id of the job to get.
+        :return: The job.
         """
         ...
 
-    @webmethod(route="/eval/benchmarks/{benchmark_id}/jobs/{job_id}/result", method="GET")
-    async def job_result(self, benchmark_id: str, job_id: str) -> EvaluateResponse:
-        """Get the result of a job.
+    @webmethod(route="/eval/jobs/{job_id}", method="DELETE")
+    async def delete_eval_job(self, job_id: str) -> Optional[EvalJob]:
```

Comment:
delete should just return `nil` I think, no? It's unusual to return a payload on DELETE, in my experience. What do you think?

Additionally, the same comment about `Optional` would apply here: when do you return `nil`+200 and when do you return `EvalJob`+200? (Both are allowed by this definition.)

Reply:
it's helpful to include the deleted resource in the response, it may have changed since it was last queried.

response 200 - include the resource
response 204 - resource not included

Reply:
Perhaps! (I don't think there's a hard standard for these things since "REST" is not a standard.) I am concerned we don't see to have standards. E.g. agent DELETE returns nil. Would be nice to have some consistency.

Reply:
definitely! i expect the /jobs api defined here will be used for all the */jobs resources.

Reply:
I've posted https://github.com/meta-llama/llama-stack/pull/1640 for discussion; I am happy for this to go either way, so consider it a strawman PR.

Note: I think we should build some tests around api definitions that would enforce such patterns across all APIs. I may take a look into it if others confirm it would be useful.